### ISSUE TITLE
Can it be extended to multi-class classification?

### ISSUE THREAD

**Issue by sruteesh** [bdhingra/tweet2vec#7]
Hi,
Great work. I am working multi classification of tweets and if i have a training data in the form of
__label__ and __tweet__ in each line, can tweet2vec be used?
I feel since the core idea is the same it should. Any thoughts?

**Comment by bdhingra** [comment]
Hi,
Thanks! Are you talking about multi-label classification? For tweet2vec, we simply split tweets with multiple labels into multiple lines during training (i.e. we repeated the tweet with each label). At test time, we used the output probabilities to come up with a ranked list of label predictions and computed precision/recall from there.

A more principled way to do this, however, would be to change the loss function during training to something that respects multiple labels (a bunch of binary classifiers, for example). You can make these changes in char.py. The core model for learning tweet representations would remain unchanged though.

Best,
Bhuwan

**Comment by anujgupta82** [comment]
@bdhingra : I guess @sruteesh 's question is related  to applying this technique to multi class classification : 
there are say 10 classes labels [1....10]. each tweet is of form <(unique) class label> - <tweet text> 

can tweet2vec approach be used ?

**Comment by bdhingra** [comment]
That is exactly what tweet2vec does currently, no? During training the tweets are preprocessed such that they have exactly one label (hashtag). During test, we consider multiple tags and compute precision/recall, but you could change test_char.py to keep only the top prediction and compute accuracy instead.

**Comment by sruteesh** [comment]
Thanks @bdhingra and @anujgupta82 for replying. Yes I am talking about multi class classification as described by @anujgupta82. I understand that tweet2vec currently does the same thing. What i want is has this approach been compared to any state of art methods and also can u state the results if applied to any gold standard data.
Thanks

**Comment by bdhingra** [comment]
This was tried on the task of predicting hashtags associated with a tweet, as explained in the [original paper](http://www.cs.cmu.edu/~bdhingra/papers/tweet2vec.pdf).
